linked_list/linked_list.py

```diff
@@ -13,63 +13,140 @@ def __init__(self):
 
     # returns the value in the first node
     # returns None if the list is empty
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)?
+    # Space Complexity: O(1)?
     def get_first(self):
-        pass
+        if self.head == None:
+            return None
 
+        return self.head.value
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
     # Time Complexity: ?
     # Space Complexity: ?
     def add_first(self, value):
-        pass
+        self.head = Node(value, self.head)
 
     # method to find if the linked list contains a node with specified value
     # returns true if found, false otherwise
     # Time Complexity: ?
     # Space Complexity: ?
     def search(self, value):
-        pass
+        current = self.head 
+
+        while current != None:

```

Comment:
We can shorten this to:

```suggestion
        while current:
```